packages/server/lib/cloud/api.ts

```diff
@@ -473,12 +474,40 @@ module.exports = {
 
       const preflightBaseProxy = apiUrl.replace('api', 'api-proxy')
 
+      const detectUrl = () => {
+        let envUrls = process.env.CYPRESS_ENV_URLS
+
+        if (envUrls) {
+          const pkgToUrls = JSON.parse(base64Url.decode(envUrls))
+
+          const key = Object.keys(pkgToUrls).find((key) => {
+            try {
+              require.resolve(key, { paths: [process.cwd()] })
```

Comment:
i dont think you can use `process.cwd()` here because it will point to `packages/server` (I think) - we may actually change `process.chdir()` when you open a project, but I know for sure we could pass in the `projectPath` into here because we have it as a discrete value from `modes/record` or `modes/run` that may be easier to pass in

Reply:
Updated here: [`0b90a2b` (#25837)](https://github.com/cypress-io/cypress/pull/25837/commits/0b90a2b68b6cfdb810cf5134d8938af4b37ec6f6)